src/components/pickers/RangeSlider/Slide.tsx

```diff
@@ -0,0 +1,34 @@
+import { SliderState } from '@react-stately/slider';
+import { useMemo } from 'react';
+
+import { StyledSlide, StyledSlideTrack } from './styled';
+
+export type SlideProps = {
+  state: SliderState;
+  ranges: number[];
+  isDisabled?: boolean;
+  isSingle?: boolean;
+};
+
+export function Slide(props: SlideProps) {
+  const { ranges, state, isDisabled, isSingle } = props;
+
+  const styles = {};
+  const mods = useMemo(
+    () => ({
+      single: isSingle,
+      disabled: isDisabled,
+    }),
+    [isDisabled, isSingle],
+  );
+
+  ranges.forEach((rangeIndex) => {
+    styles[`@thumb-${rangeIndex}-value`] = state.getThumbPercent(rangeIndex);
```

Comment:
Use `style` property for such dynamic custom properties. It won't hurt performance that way.